### PR TITLE
remove suffixes from generated StorageClasses and VolumeSnapshotClass…

### LINKED ISSUE #96234: Clean up / remove suffixes from generated StorageClasses and VolumeSnapshotClasses in E2E tests
**What would you like to be added**:
Suffixes for generated `StorageClass` and `VolumeSnapshotClass` objects can be removed since they already have randomly generated names.
https://github.com/kubernetes/kubernetes/blob/396b90f06ca29c754a81eb6c03cff50b7d9bf265/test/e2e/storage/testsuites/driveroperations.go#L76

**Why is this needed**:
This is a small cleanup to remove an argument that isn't needed (it's hardcoded in callers) and / or repeats the name of the snapshotter in the suffix which can reduce readability:
https://github.com/kubernetes/kubernetes/blob/396b90f06ca29c754a81eb6c03cff50b7d9bf265/test/e2e/storage/drivers/csi.go#L163-L170

/sig storage

### PR DESCRIPTION
**Pull request by lala123912**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Remove redundant suffix to improve readability.
**Which issue(s) this PR fixes**:
Fixes #96234

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

### CODE REVIEW

**Comment by chrishenzie** [comment]
/lgtm

@msau42

**Review by chrishenzie — APPROVED** [review]

File: test/e2e/storage/testsuites/driveroperations.go [hunk 1]
```diff
@@ -73,8 +72,7 @@ func GetStorageClass(
 		},
 		ObjectMeta: metav1.ObjectMeta{
 			// Name must be unique, so let's base it on namespace name and use GenerateName
-			// TODO(#96234): Remove unnecessary suffix.
-			Name: names.SimpleNameGenerator.GenerateName(ns + "-" + suffix),
+			Name: names.SimpleNameGenerator.GenerateName(ns),
```
Comment (msau42) [inline test/e2e/storage/testsuites/driveroperations.go:75]: It looks like some usages were using the suffix to append the driver name to the field to make it easier to tell which driver is being tested. Can we retain the same behavior by getting it automatically from the passed in storageclass and snapshotclass?

Comment (lala123912) [reply test/e2e/storage/testsuites/driveroperations.go:75]: > It looks like some usages were using the suffix to append the driver name to the field to make it easier to tell which driver is being tested. Can we retain the same behavior by getting it automatically from the passed in storageclass and snapshotclass?

According to the tests calling this function, the member "Provisioner" or "driver" is the driver name so we can get it from them.

**Comment by lala123912** [comment]
see the new pr https://github.com/kubernetes/kubernetes/pull/97183.

**Pull request closed without merging**